### PR TITLE
Use metis-feedstock

### PR DESCRIPTION
**Pull request by basnijholt**
Addresses #17 

[CircleCI build on my channel](https://circleci.com/gh/basnijholt/suitesparse-feedstock/2) is passing
[Travis build on my channel](https://travis-ci.org/basnijholt/suitesparse-feedstock/builds/191277026) is passing

### CODE REVIEW

**Comment by jakirkham** [comment]
Should we be pinning `metis`?

**Comment by jakirkham** [comment]
Could you please take a look at this when you have a chance, @grlee77?

**Comment by grlee77** [comment]
I think this is probably a good change to make.  

I am not sure if the same approach can be used on the 4.4.6 branch .  We would have to try it and see if the libraries are found or not.

The version of METIS packaged with SuiteSparse 4.5.3 is METIS 5.1.0 so we should probably pin to a version of METIS no older than that if not that exact version.

For SuiteSparse 4.4.6 it looks like the packaged METIS is 4.0.1.

**Comment by basnijholt** [comment]
I have pinned metis to 5.1.0.

In the old recipe both static and dynamic libs of Metis were build. Is this now also necessary?

**Comment by jakirkham** [comment]
> I am not sure if the same approach can be used on the 4.4.6 branch . We would have to try it and see if the libraries are found or not.

It wouldn't unfortunately. 😒 4.4.6 requires an older version METIS that has licensing issue. That said, it doesn't build an internal copy of METIS either. So it shouldn't present a problem the way that 4.5.x is.

**Comment by jakirkham** [comment]
> I have pinned metis to 5.1.0.

> In the old recipe both static and dynamic libs of Metis were build. Is this now also necessary?

If we can use shared libraries from Metis, we might be able to be more flexible with the pinning. However, if we use static libraries, we probably should pin exactly. How is this configured currently?

**Comment by basnijholt** [comment]
The current metis-feedstock build are shared libraries.

**Comment by grlee77** [comment]
The other two packages I worked on that depend on `suitesparse` are `scikit-umfpack` and `scikit-sparse` and I believe those both link to the shared libs.

**Comment by basnijholt** [comment]
OK great! Ready to merge then?

**Comment by grlee77** [comment]
As far as I can tell `cvxopt` is the only other package aside from the ones I mentioned above that is using `metis`.  It is pinned to `metis` 4.4.x, so not affected by this PR either.

I am +1 on merging.  If interested, can you please add yourself to the maintainers section of `meta.yaml`?

**Comment by grlee77** [comment]
I corrected the prior comment to say `suitesparse` rather than `metis`.

**Comment by basnijholt** [comment]
OK great, so let's merge! 😊

About adding me as a package maintainer: I don't think I know enough about suitesparse to qualify as maintainer.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,11 +28,13 @@ requirements:
     - openblas 0.2.19|0.2.19.*
     - tbb
     - cmake
+    - metis ==5.1.0
```
Comment (jakirkham) [inline recipe/meta.yaml]: Do we need this strict of a pinning if we are using shared libraries?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -28,11 +28,13 @@ requirements:
...
 
   run:
     - blas 1.1 {{ variant }}
     - openblas 0.2.19|0.2.19.*
     - tbb
+    - metis ==5.1.0
```
Comment (jakirkham) [inline recipe/meta.yaml]: Same as above.

**Comment by jakirkham** [comment]
> About adding me as a package maintainer: I don't think I know enough about suitesparse to qualify as maintainer.

But you knew enough to raise this issue... 😉

**Comment by jakirkham** [comment]
Once we have settled the pinning, it should be added to the [pinning script]( https://github.com/conda-forge/conda-forge.github.io/blob/2c78ebc8eec2db4d55889bfd16c429cf3edac652/scripts/pin_the_slow_way.py#L50-L51 ).

File: recipe/meta.yaml (hunk 1 above)
Comment (basnijholt) [inline recipe/meta.yaml]: Well `suitesparse` 4.5.3 uses this version, so it's just to be safe I guess?

**Comment by grlee77** [comment]
It looks like the METIS 5.1.0 is from 2013 so apparently the library is not updated often.  I would recommend just pinning to 5.1.* for now.  We can always update the pinning again later if needed.

I don't think any of us are experts on on SuiteSparse, so I wouldn't hold back joining as a maintainer on that account.  It is perfectly okay if you are not interested, though.

**Comment by basnijholt** [comment]
I have pinned it to `5.1.*` and due to popular demand, I have added myself as maintainer :)

File: recipe/meta.yaml [hunk 3]
```diff
@@ -28,11 +28,13 @@ requirements:
     - openblas 0.2.19|0.2.19.*
     - tbb
     - cmake
+    - metis ==5.1.*
 
   run:
     - blas 1.1 {{ variant }}
     - openblas 0.2.19|0.2.19.*
     - tbb
+    - metis ==5.1.*
```
Comment (jakirkham) [inline recipe/meta.yaml]: FYI don't think the `==` is necessary. `metis 5.1.*` works too and is a bit nicer looking. 😉

**Comment by jakirkham** [comment]
Pinning `metis` globally in PR ( https://github.com/conda-forge/conda-forge.github.io/pull/315 ).

**Pull request merged**